src/actions/actions.ts

```diff
@@ -5287,3 +5288,287 @@ class ActionOverrideCmdAltUp extends BaseCommand {
+    keys = ["\\", "\\", "w"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words after the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        min: position
+      });
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionEndForwardCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "e"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the end of all words after the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        min: position
+      });
+    }
+
+    public getMatchPosition(match: EasyMotion.Match, position: Position, vimState: VimState): Position {
+        return new Position(match.position.line, match.position.character + match.text.length - 1);
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionEndBackwardCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "g", "e"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words before the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        max: position,
+      });
+    }
+
+    public getMatchPosition(match: EasyMotion.Match, position: Position, vimState: VimState): Position {
+        return new Position(match.position.line, match.position.character + match.text.length - 1);
+    }
+  }
+
+  @RegisterAction
+  class ActionEasyMotionBeginningWordCommand extends BaseEasyMotionCommand {
+    modes = [ModeName.Normal];
+    keys = ["\\", "\\", "b"];
+
+    public getMatches(position: Position, vimState: VimState): EasyMotion.Match[] {
+      // Search for the beginning of all words before the cursor
+      return vimState.easyMotion.sortedSearch(position, "\\w{1,}", {
+        isRegex: true,
+        max: position,
+      });
+    }
+  }
+
+  @RegisterAction
+  class CommandEscEasyMotionMode extends BaseCommand {
+    modes = [
+      ModeName.EasyMotionMode
+    ];
+    keys = [
+      ["<Esc>"],
+      ["<C-c>"],
+      ["<C-[>"],
+    ];
+
+    public async exec(position: Position, vimState: VimState): Promise<VimState> {
+      // Escape or other termination keys were pressed, exit mode
+      vimState.currentMode = ModeName.Normal;
+      vimState.easyMotion.exitMode();
+
+      return vimState;
+    }
+  }
+
+  @RegisterAction
+  class MoveEasyMotion extends BaseMovement {
+    modes = [ModeName.EasyMotionMode];
+    keys = ["<character>"];
+
+    public async execAction(position: Position, vimState: VimState): Promise<Position> {
+      var key = this.keysPressed[0];
+      if (!key) {
+        return position;
+      }
+
+      // "nail" refers to the accumulated depth keys
```

Comment:
I honestly have no idea, I couldn't find a decent word to describe what it does, which is why I am referring to it as "accumulated depth keys". Got a better name? :)